README.qmd

```diff
@@ -446,9 +452,814 @@ rm(x)
+
+- Tibbles never coerce vectors. 
+  - A common need is to suppress in `data.frame()` by setting `stringsAsFactors` to `FALSE`
+- Tibbles surround non-syntactic names with `` ` `` rather than transforming them
+- Data frames recycle inputs that are an integer multiple of the longest vector, while tibbles only recycle vectors that are of length 1
+- Tibbles allow you to refer to created variables during construction
+<br><br>
+
+#### Row names
+
+A character vector can be supplied to label the "rows" of a data frame, in two ways:
+
+- The `row.names` argument in `data.frame()`
+- By calling `rownames()`
+
+Tibbles do not support row names for three main reasons
+- They are stored differently from data
+- They only work when rows can be identified via a single string
+- They *must* be unique, so repetition/resampling results in new row names
+
+Row names can be converted to a column in a tibble using `rownames_to_column()` or the `rownames` argument in `as_tibble()`
+<br><br>
+
+#### Printing
+
+Four main differences between the `print()` output for a data frames and for tibbles:
+
+- Tibbles show the first 10 rows and all columns that fit on screen
+- Columns are labelled with its abbreviated type
+- Wide columns are truncated
+- Color is used when supported to (de)emphasize information
+<br><br>
+
+#### Subsetting
+
+Discussed more in Chapter 4, you can subset data frames and tibbles like a 1-D list or a 2-D matrix.
+
+Tibbles modify two undesirable properties of data frames:
+
+- Data frames will return a vector for `df[, x]` if x is a length one vector, and a data frame is x is of length > 1. To avoid this, use `df[, x, drop = FALSE]`
+  - Subsetting a single column from a tibble using `[` can cause an issue with legacy code that uses `df[, "col"]` to return a column as an atomic vector. Using `df[["col"]]` returns the desired vector whether subsetting a data frame or tibble
+```{r}
+df1 <- data.frame(xyz = "a")
+df2 <- tibble(xyz = "a")
+x <- "xyz"
+df1[, x]
+df1[, x, drop = FALSE]
+df2[, x]
+```
+<br>
+
+- When using `$` data frames will return any variable that starts with the input
+```{r}
+str(df1$x)
+str(df2$x)
+```
+<br>
+
+#### Testing and coercing
+
+- Check with `is.data.frame()` or `is_tibble()`
+- Coerce with `as.data.frame()` or `as_tibble()`
+<br><br>
+
+#### List columns
+
+Since data frames are lists of vectors, a data frame can have a column as a list.
+
+Adding a list column involves an extra step in data frames:
+```{r}
+# Either after the data frame is created
+d <- data.frame(a = 1:3)
+d$b <- list(4:6)
+```
+<br>
+
+Lists are fully supported in tibbles
+```{r}
+d <- tibble(
+  a = 1:3,
+  b = list(4:6)
+)
+```
+<br>
+
+#### Matrix and data frame columns
+
+Extending the length requirement for data frames, it's actually the `NROW()` of each column that needs to match, not the `length()`. Because of this, data frames and matrices can be included as columns in a data frame.
```

Comment:
```suggestion
Extending the length requirement for data frames (that all columns must be of the same length), it's actually the `NROW()` of each column that needs to match. Because of this, data frames and matrices can be included as columns in a data frame.
```